### PR TITLE
Add devise

### PR DESCRIPTION
**Pull request by the-faizmohammad**
1. Removed the current_user method from the ApplicationController as we can implement the same from devise gem.
2. Installed the devise gem to implement the login and signup actions.
3. Used rails generate devise User to implement the devise for the User model.
4. Used the rails generate devise:views to generate the views related to the signup, and login authentication action views.
5. Updated the ActionController with before_with to implement user authentications and also defined strong parameters.

image

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,4 +1,12 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable, :confirmable
```
Comment (mckent05) [inline app/models/user.rb]: - You have done well so far, but you seem to have setup two devise module in your user model, I suggest you delete one, (the one without the `confirmable` module). Please fix this 🙏

File: db/migrate/20240202213724_add_devise_to_users.rb [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      ## Rememberable
+      t.datetime :remember_created_at
+
+      ## Trackable
+      # t.integer  :sign_in_count, default: 0, null: false
+      # t.datetime :current_sign_in_at
+      # t.datetime :last_sign_in_at
+      # t.string   :current_sign_in_ip
+      # t.string   :last_sign_in_ip
+
```
Comment (mckent05) [inline db/migrate/20240202213724_add_devise_to_users.rb]: - Good job done so far, but I have an issue running your migrations, please fix this issue and ensure all migrations run without error.
wagegawega

**Review by mckent05 — CHANGES_REQUESTED** [review]
Hi @the-faizmohammad, 

# STATUS: CHANGES REQUIRED ♻️ ♻️ 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
almost there

## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👍 👍 
- Good job adding devise to be your application 👍 👏 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Ibizugbe — COMMENTED** [review]
Hi @the-faizmohammad 👋🏼👋🏼👋🏼, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

I can see you are probably repeating this milestone. since a review has been placed previously and it wasn't approved, kindly work on a fresh branch, then open a pull request that will then be submitted👍🏼. Also, kindly note that since this is a pair programming activity, the active contributions of your current partner this week are required. kindly reach out to your partner and work with Him/Her. If you've been permitted to work alone this week, then add a detailed screenshot for this week from student success permitting you to work alone, and attach it to the description of the new PR.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Pull request merged**